packages/ui/EnvironmentNotice/index.tsx

```diff
@@ -28,7 +18,7 @@ export default async function EnvironmentNotice({ environmentId }: EnvironmentNo
         <p>
           {`You're currently in the ${environment.type} environment.`}
           <a
-            href={replaceEnvironmentId(currentUrl, otherEnvironmentId)}
+            href={`${WEBAPP_URL}/environments/${otherEnvironmentId}/settings/api-keys`}
```

Comment:
`EnvironmentNotice` is a universal component used in different places; that's why we kept the current URL and only replaced the environmentId in this.
The component is also used in e.g. `apps/web/app/(app)/environments/[environmentId]/settings/setup/page.tsx`

Please find an approach, that works with every URL and replaces the environmentId in the url.

Reply:
Alternatively we can also pass the subPageUrl in the params. Not so cool, but if we can't find another approach that would also work